asllib/Typing.ml

```diff
@@ -29,7 +29,7 @@ module TypingRule = Instrumentation.TypingRule
 let ( |: ) = Instrumentation.TypingNoInstr.use_with
 let fatal_from = Error.fatal_from
 let undefined_identifier pos x = fatal_from pos (Error.UndefinedIdentifier x)
-let unsupported_expr e = fatal_from e (Error.UnsupportedExpr e)
+let unsupported_expr e = fatal_from e Error.(UnsupportedExpr (Static, e))
```

Comment:
To suppress two "this open statement shadows the value identifier unsupported_expr...", I am in favor of deleting this line and of using the fully qualified reference `StaticInterpreter.unsupport_expr` where appropriate, which clearly states that a "static" error is flagged.
```suggestion
```

Reply:
Notice that there are three occurences of `unsupported_expr` that need to be qualified.